doc/source/dev/roadmap-detailed.rst

```diff
@@ -71,49 +59,67 @@ add new benchmarks, however running the benchmarks is not very intuitive.
+`gh-13615 <https://github.com/scipy/scipy/issues/13615>`_.
+
+
 Use of Cython
 `````````````
-Regarding Cython code:
-
-- It's not clear how much functionality can be Cythonized without making the
-  .so files too large.  This needs measuring.
-- Cython's old syntax for using NumPy arrays should be removed and replaced
-  with Cython memoryviews.
+Cython's old syntax for using NumPy arrays should be removed and replaced with
+Cython memoryviews. When Cython 3.0 is released, the last use of the deprecated
+NumPy C API (by Cython, everything in SciPy was fixed) will disappear. Then we
+can define ``NPY_NO_DEPRECATED_API`` unconditionally.
 
 
-Windows build issues
-````````````````````
-SciPy critically relies on Fortran code. This is still problematic on Windows.
-There are currently only two options: using Intel Fortran, or using MSVC +
-gfortran.  The former is expensive, while the latter works (it's what we use
-for releases) but is quite hard to do correctly.  For allowing contributors and
-end users to reliably build SciPy on Windows, using the Flang compiler looks
-like the best way forward long-term.
+Use of Pythran
+``````````````
+Pythran is still an optional build dependency, and can be disabled with
+``SCIPY_USE_PYTHRAN=0``. The aim is to make it a hard dependency - for that to
+happen it must be clear that the maintenance burden is low enough (Meson will
+help here, because it removes the monkey patching that is now done to enable
+Pythran).
 
 
 Continuous integration
 ``````````````````````
-Continuous integration is in good shape, it currently covers the Windows, macOS
-and Linux, ARM64 and ppc64le platforms, as well as a range of versions of our
-dependencies and building release quality wheels.
+Continuous integration currently covers 32/64-bit Windows, macOS on x86-64, and
+32/64-bit Linux on x86, and Linux on aarch64 - as well as a range of versions
+of our dependencies and building release quality wheels. Reliability of CI has
+not been good recently (H2 2021), due to the large amount of configurations to
+support and some CI jobs needing an overhaul. We aim to reduce build times,
+improve caching, move more jobs to GitHub Actions, drop TravisCI and Appveyor
+in the `scipy-wheels repo <https://github.com/MacPython/scipy-wheels>`_,
+move from ``multibuild`` to ``cibuildwheel`` for building wheels for releases,
+and make the set of configurations in CI jobs more orthogonal.
 
 
 Size of binaries
 ````````````````
-SciPy binaries are quite large (e.g. an unzipped manylinux wheel for 1.4.1 is
-91 MB), and this can be problematic - for example for use in AWS Lambda, which
-has a 250 MB size limit. We aim to keep binary size as low as possible; when
-adding new compiled extensions, this needs checking. Stripping of debug symbols
-in ``multibuild`` can likely be improved (see `this issue
-<https://github.com/matthew-brett/multibuild/issues/162>`__).
+SciPy binaries are quite large (e.g. an unzipped manylinux wheel for 1.7.3 is
```

Comment:
Was there any discussions about making some part of SciPy optional to install? Maybe something like a SciPy-core.

Reply:
There was, at various points. It's not all that interesting though it turns out, because some of the largest modules with the most build-related complexity are the ones you can't really split off (`linalg`, `special`). `stats` is probably the one exception - but that's kind of an important module as well.